bin/pkg/validate_lock_dir.ml

```diff
@@ -0,0 +1,79 @@
+open! Import
+open Pkg_common
+module Package_universe = Dune_pkg.Package_universe
+module Lock_dir = Dune_pkg.Lock_dir
+module Opam_repo = Dune_pkg.Opam_repo
+module Package_version = Dune_pkg.Package_version
+module Opam_solver = Dune_pkg.Opam_solver
+
+let info =
+  let doc = "Validate that a lockdir contains a solution for local packages" in
+  let man = [ `S "DESCRIPTION"; `P doc ] in
+  Cmd.info "validate-lockdir" ~doc ~man
+;;
+
+let enumerate_lock_dirs_by_path ~context_name_arg ~all_contexts_arg =
+  let open Fiber.O in
+  let+ per_contexts =
+    Per_context.choose ~context_name_arg ~all_contexts_arg ~version_preference_arg:None
+  in
+  List.filter_map per_contexts ~f:(fun { Per_context.lock_dir_path; _ } ->
+    if Path.exists (Path.source lock_dir_path)
+    then (
+      try Some (lock_dir_path, Lock_dir.read_disk lock_dir_path) with
+      | User_error.E e ->
+        User_warning.emit
```

Comment:
If the lock dir is to be validated and it can't be read, should this be a warning? It seems like an error.

Reply:
The problem is that if this command is run with `--all-contexts` I still want it to detect as many invalid lockdirs as it can. I'll change this section to treat a lockdir that can't be parsed as the same type of error as an invalid lockdir.